spec/unit/functions/lookup_spec.rb

```diff
@@ -812,9 +812,9 @@ def explain(key, options = {})
           YAML
         end
 
-        it 'fails lookup and reports a type mismatch' do
+        it 'fails lookup and reports unspecified class', :if => Gem::Version.new(RUBY_VERSION.dup) >= Gem::Version.new('2.1.0') do
           expect { lookup('a') }.to raise_error do |e|
-            expect(e.message).to match(/key 'a'.*data_hash function 'yaml_data'.*using location.*wrong type, expects Puppet::LookupValue, got Runtime/)
+            expect(e.message).to match(/Tried to load unspecified class: Puppet::Graph::Key/)
```

Comment:
What is the behavior on Ruby < 2.1?

Reply:
I pushed a commit where the old test i reinstated with the condition to run only if Ruby is < 2.1.